makefiles/instance/rules.mk

```diff
@@ -124,6 +134,24 @@ else
 TARGET_ARCHS = $(ARCHS)
 endif
 
+_TARGET_ARCHS_COUNT = $(words $(TARGET_ARCHS))
+
+# The magic behind `make commands`: proxy the compile commands for one arch
+# through gen-commands.pl so that we can save them to the commands db
+export THEOS_GEN_COMPILE_COMMANDS
+ifeq ($(THEOS_GEN_COMPILE_COMMANDS) $(THEOS_CURRENT_ARCH),$(_THEOS_TRUE) $(firstword $(TARGET_ARCHS)))
+_TARGET_SWIFTC := $(TARGET_SWIFTC)
+# we use PWD instead of THEOS_PROJECT_DIR because the latter always refers to the root level project
+# so it isn't correct for subprojects.
```

Comment:
Do we need the `TARGET_SWIFTC` and `TARGET_CXX` vars to be re-evaluated each call? If not, we might want to consider changing them to simple assigns (`:=`) so that we can do `TARGET_CXX := <more stuff here> $(TARGET_CXX)`  or `TARGET_CXX := $(addprefix <some text>, $(TARGET_CXX))`